proto/api.proto

```diff
@@ -241,6 +241,25 @@ message ScanResponse {
   repeated KeyValue rows = 2 [(gogoproto.nullable) = false];
 }
 
+// A ReverseScanRequest is the argument to the ReverseScan() method. It specifies the
+// start and end keys for the scan and the maximum number of results.The InclusiveKey
+// and InclusiveEndKey flags are used to indicate whether the results need to include
+// the start key and end key.
+message ReverseScanRequest {
+  optional RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Must be > 0.
+  optional int64 max_results = 2 [(gogoproto.nullable) = false];
+  optional bool inclusive_key = 3 [(gogoproto.nullable) = false];
+  optional bool inclusive_end_key = 4 [(gogoproto.nullable) = false];
```

Comment:
Shouldn't inclusive_end_key always be true? I'm not sure it makes sense to provide this option. Also, since we'll end up creating default go structs for the reserve scan request, this will be false by default. That definitely won't give expected results. I think this should just be removed.

Also, seems like maybe we should rename "inclusive_key" to "inclusive_start_key".